src/package.cc

```diff
@@ -25,4 +25,29 @@ Package::Package() : id_(next_id_++), fill_min_(0), fill_max_(std::numeric_limit
 
 Package::Package(std::string name, double fill_min, double fill_max, std::string strategy) : name_(name), id_(next_id_++), fill_min_(fill_min), fill_max_(fill_max), strategy_(strategy) {}
 
+double GetFillMass(Resource::Ptr r, Package::Ptr pkg) {
+  if (r->quantity() < pkg->fill_min()) {
+    // less than one pkg of material available
+    return 0;
+  }
+
+  std::vector<Resource::Ptr> rs;
+  Resource::Ptr r_pkgd;
+  double fill_mass;
+  if (pkg->strategy() == "first") {
+    fill_mass = pkg->fill_max();
+  } else if (pkg->strategy() == "equal") {
+    int num_min_fill = std::floor(r->quantity() / pkg->fill_min());
+    int num_max_fill = std::ceil(r->quantity() / pkg->fill_max());
+    if (num_min_fill >= num_max_fill) {
+      // all material can fit in a package
+      double fill_mass = r->quantity() / num_max_fill;
```

Comment:
Can you walk me through the logic of this one? Starting with what the `equal` strategy is supposed to be.

Reply:
`equal` is a strategy that attempts to fill as many packages as necessary with an equal inventory.  This strategy does have the case where distributing a certain quantity equally can result in underfilling the packages and using one fewer packages is not enough inventory for all the quantity requested.  In response to a similar comment by me @nuclearkatie gave an example.  It may be useful to have that example in a comment.

Reply:
See header  49 to 62 now:
```
/// Returns optimal fill mass for a resource to be packaged. Can be used
/// to determine how to respond to requests for material, and to actually
/// package and send off trades.
/// Packaging strategy "first" simply fills the packages one by one to the
/// maximum fill. Therefore, it should always try to max fill.
/// Packaging strategy "equal" tries to fill all packages to the same mass.
/// This tries to find the optimal number and fill mass of packages given the
/// packaging limitations. It does this by calculating bounding fills, 
/// floor(quantity/fill_min) and ceiling(quantity/fill_max). 
/// There might be a scenario where there is no solution, i.e. an integer
/// number of packages cannot be filled with no remainder. In this case,
/// the most effective fill strategy is to fill to the max. Numeric example:
/// quantity = 5, fill_min = 3, fill_max = 4. num_min_fill = floor(5/3) = 1,
/// num_max_fill = ceil(5/4) = 2. num_min_fill < num_max_fill, so fill to max.
```